shared/src/metabase/mbql/schema.cljc

```diff
@@ -44,139 +32,108 @@
 ;;    the future we will likely add middleware that uses this metadata to automatically validate that a driver has the
 ;;    features needed to run the query in question.
 
-;;; +----------------------------------------------------------------------------------------------------------------+
-;;; |                                                  MBQL Clauses                                                  |
-;;; +----------------------------------------------------------------------------------------------------------------+
+(def ^:private NonBlankString
+  [:ref ::lib.schema.common/non-blank-string])
+
+(def ^:private BaseType
+  [:ref ::lib.schema.common/base-type])
+
+(def ^:private SemanticOrRelationType
+  [:ref ::lib.schema.common/semantic-or-relation-type])
+
+(def ^:private PositiveInt
+  [:ref ::lib.schema.common/positive-int])
+
+(def ^:private IntGreaterThanOrEqualToZero
+  [:ref ::lib.schema.common/int-greater-than-or-equal-to-zero])
 
-;;; ------------------------------------------------- Datetime Stuff -------------------------------------------------
+(def ^:private FieldID
+  [:ref ::lib.schema.id/field])
+
+(def ^:private CardID
+  [:ref ::lib.schema.id/card])
+
+(def ^:private TableID
+  [:ref ::lib.schema.id/table])
+
+(def ^:private DateLiteral
+  [:ref ::lib.schema.literal/date])
+
+(def ^:private DateTimeLiteral
+  [:ref ::lib.schema.literal/datetime])
+
+(def ^:private TimeLiteral
+  [:ref ::lib.schema.literal/time])
 
 ;; `:day-of-week` depends on the [[metabase.public-settings/start-of-week]] Setting, by default Sunday.
 ;; 1 = first day of the week (e.g. Sunday)
 ;; 7 = last day of the week (e.g. Saturday)
-(def date-bucketing-units
+(def ^:private date-bucketing-units
   "Set of valid units for bucketing or comparing against a *date* Field."
   #{:default :day :day-of-week :day-of-month :day-of-year :week :week-of-year
     :month :month-of-year :quarter :quarter-of-year :year})
```

Comment:
Yeah, it is more or less the same as what's in that namespace. I went back and forth on how far to try to go in consolidating things, but I eventually decided I should try to keep the version in `metabase.mbql.schema` mostly separate from the new version in `metabase.lib.schema`. That way if we make further changes (for example, add support for a new temporal unit) we're not accidentally unintentionally changing the schema for legacy queries. 

Once we migrate all the QP code over we can just delete this namespace entirely, at which point all the duplication will be fixed

Reply:
I did consolidate some stuff, such as the schemas for temporal literal strings like `2023-08-03` for a date literal, since that isn't something that is possibly going to change in the future. Things that could possibly like the set of temporal units I left separate